x-pack/plugins/observability/public/context/has_data_context.tsx

```diff
@@ -0,0 +1,123 @@
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { uniqueId } from 'lodash';
+import React, { createContext, useEffect, useState } from 'react';
+import { Alert } from '../../../alerts/common';
+import { getDataHandler } from '../data_handler';
+import { FETCH_STATUS } from '../hooks/use_fetcher';
+import { usePluginContext } from '../hooks/use_plugin_context';
+import { useTimeRange } from '../hooks/use_time_range';
+import { getObservabilityAlerts } from '../services/get_observability_alerts';
+import { ObservabilityFetchDataPlugins, UXHasDataResponse } from '../typings/fetch_overview_data';
+
+type DataContextApps = ObservabilityFetchDataPlugins | 'alert';
+
+export type HasDataMap = Record<
+  DataContextApps,
+  { status: FETCH_STATUS; hasData?: boolean | UXHasDataResponse | Alert[] }
+>;
+
+export interface HasDataContextValue {
+  hasData: Partial<HasDataMap>;
+  hasAnyData: boolean;
+  isAllRequestsComplete: boolean;
+  onRefreshTimeRange: () => void;
+  forceUpdate: string;
+}
+
+export const HasDataContext = createContext({} as HasDataContextValue);
+
+const apps: DataContextApps[] = ['apm', 'uptime', 'infra_logs', 'infra_metrics', 'ux', 'alert'];
+
+export function HasDataContextProvider({ children }: { children: React.ReactNode }) {
+  const { core } = usePluginContext();
+  const [forceUpdate, setForceUpdate] = useState('');
+  const { absStart, absEnd } = useTimeRange();
+
+  const [hasData, setHasData] = useState<HasDataContextValue['hasData']>({});
+
+  useEffect(
+    () => {
+      apps.forEach(async (app) => {
+        try {
+          if (app !== 'alert') {
+            const params =
+              app === 'ux' ? { absoluteTime: { start: absStart, end: absEnd } } : undefined;
+
+            const result = await getDataHandler(app)?.hasData(params);
+            setHasData((prevState) => ({
+              ...prevState,
+              [app]: {
+                hasData: result,
+                status: FETCH_STATUS.SUCCESS,
+              },
+            }));
+          }
+        } catch (e) {
+          setHasData((prevState) => ({
+            ...prevState,
+            [app]: {
+              hasData: undefined,
+              status: FETCH_STATUS.FAILURE,
+            },
+          }));
+        }
+      });
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    []
+  );
+
+  useEffect(() => {
+    async function fetchAlerts() {
+      try {
+        const alerts = await getObservabilityAlerts({ core });
+        setHasData((prevState) => ({
+          ...prevState,
+          alert: {
+            hasData: alerts,
+            status: FETCH_STATUS.SUCCESS,
+          },
+        }));
+      } catch (e) {
+        setHasData((prevState) => ({
+          ...prevState,
+          alert: {
+            hasData: undefined,
+            status: FETCH_STATUS.FAILURE,
+          },
+        }));
+      }
+    }
+
+    fetchAlerts();
```

Comment:
Is it the right place to fetch alerts in the `HasDataContextProvider`?

Reply:
I thought would be nice to centralize all requests to fill the page in a single place. Alert would have behaved in the same way as the other apps if I registered it. Maybe that is the change I should do.